src/transaction.ts

```diff
@@ -11,13 +11,22 @@ export class Transaction implements IModel  {
   public amount: number;
   public date: Date;
   public signature: string;
+  public smartContractId: number;
+  public contractData: string;
 
-  constructor(type: TransactionType, recipientKey: string,
-              senderKey: string, amount: number, date: Date) {
+  constructor(type: TransactionType, recipientKey: string, senderKey: string, 
+              amount: number, date: Date, smartContractId?: number, contractData?: string) {
     this.recipientKey = recipientKey;
     this.amount = amount;
     this.date = date;
     this.type = type;
     this.senderKey = senderKey;
+    
+    if (smartContractId != undefined) {
```

Comment:
This check is unnecessary.

Either mark the field/ property optional as well or use the `or` operator.

```
this.smartContractId = smartContractId || 0;
```

Reply:
Can I mark a field/property as nullable by doing the following:
`privare smartContractId: string | null` ?

Reply:
you can make a field nullable by doing this: `private smartContractId?: string;`